### PR TITLE
China Lake and it's ammo can now be found

### PR DESCRIPTION
**Pull request by genessee596**
## About The Pull Request

The China Lake can now be found (extremely rarely) wherever a projectile gun can be found. It's ammunition (rubber grenade shells, explosive grenade shells, or frag grenade shells) can now be found (rarely) wherever ammunition can be found.

## Why It's Good For The Game

image
Gray and Reere told me to do this

## Changelog
:cl:
tweak: The China Lake and it's specialized ammunition should now correctly spawn in places across the ship.
/:cl:

### CODE REVIEW

**Comment by TheShown911** [comment]
You could probably make it spawn with 2 shells of it's ammo, due to it being useless on other guns.

**Comment by genessee596** [comment]
> You could probably make it spawn with 2 shells of it's ammo, due to it being useless on other guns.

Good call. I'll write that up now.

**Comment by genessee596** [comment]
It now starts loaded with 2 rubber grenade shells.

**Comment by TheShown911** [comment]
> > You could probably make it spawn with 2 shells of it's ammo, due to it being useless on other guns.
> 
> Good call. I'll write that up now.

There should be a system for tiems to spawn with other items (like voidsuits and magboots) so it shouldn't be too hard.

**Comment by TheShown911** [comment]
> It now starts loaded with 2 rubber grenade shells.

This also works, but if we ever make the gun on  design disk, or purchasable from the guild it will also spawn with 2 shells inside.

**Comment by genessee596** [comment]
> > It now starts loaded with 2 rubber grenade shells.
> 
> This also works, but if we ever make the gun on design disk, or purchasable from the guild it will also spawn with 2 shells inside.

If we do end up doing that I can just make the one that spawns in maint a subtype of the main gun that spawns with two shells while the main gun starts empty.

But I'll hold off on that for now until we need to for the sake of not risking the possibility of creating more bugs.

**Comment by TheShown911** [comment]
> > > It now starts loaded with 2 rubber grenade shells.
> > 
> > 
> > This also works, but if we ever make the gun on design disk, or purchasable from the guild it will also spawn with 2 shells inside.
> 
> If we do end up doing that I can just make the one that spawns in maint a subtype of the main gun that spawns with two shells while the main gun starts empty.
> 
> But I'll hold off on that for now until we need to for the sake of not risking the possibility of creating more bugs.

It feels like a wasted oppertunity to not use the systems we already have, and that should be as simple as a one line change.

**Comment by genessee596** [comment]
Done.

**Comment by Evankhell561** [comment]
> > > You could probably make it spawn with 2 shells of it's ammo, due to it being useless on other guns.
> > 
> > 
> > Good call. I'll write that up now.
> 
> There should be a system for tiems to spawn with other items (like voidsuits and magboots) so it shouldn't be too hard.

there is a variable called "accompanying_object"

**Comment by Evankhell561** [comment]
however all this has given me an idea to further improve the spawn values. I will add that each object has an intrinsic probability that its companion object will be generated.

**Review by TorinoFermic — APPROVED** [review]

**Comment by genessee596** [comment]
I hate to merge my own PR, but this has gotten approval from maintainers and I was asked to do it by the project lead, so it should be okay to merge.

**Pull request merged**
